From 4de033d1ee41db6254630879bbaa89657189c7d1 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Wed, 12 Dec 2018 12:06:34 +0200
Subject: [PATCH 006/269] s32g: Add dummy console flush function

Upstream code now mandates the implementation of console_flush() and
crashes if it does not exist.

Upstream-Status: Pending 

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-3452
---
 plat/s32g/include/platform_def.h | 1 +
 plat/s32g/s32g_helpers.S         | 5 +++++
 2 files changed, 6 insertions(+)

diff --git a/plat/s32g/include/platform_def.h b/plat/s32g/include/platform_def.h
index 7a2041d4d..3d29607ed 100644
--- a/plat/s32g/include/platform_def.h
+++ b/plat/s32g/include/platform_def.h
@@ -151,6 +151,7 @@ struct console_s32g {
 int console_s32g_register(uintptr_t baseaddr, uint32_t clock, uint32_t baud,
 			  struct console_s32g *console);
 int console_s32g_putc(int c, struct console_s32g *console);
+int console_s32g_flush(struct console_s32g *console);
 int xrdc_enable(void *xrdc_addr);
 #endif /* __ASSEMBLY__ */
 #endif /* PLATFORM_DEF_H */
diff --git a/plat/s32g/s32g_helpers.S b/plat/s32g/s32g_helpers.S
index b4c5683da..caa044822 100644
--- a/plat/s32g/s32g_helpers.S
+++ b/plat/s32g/s32g_helpers.S
@@ -16,6 +16,7 @@
 
 .globl console_s32g_register
 .globl console_s32g_putc
+.globl console_s32g_flush
 
 /*
  * Almost all these helper functions have been copied copied from
@@ -203,3 +204,7 @@ tx_fifo_full:
 done:
 	ret
 endfunc console_s32g_putc
+
+func console_s32g_flush
+	ret
+endfunc console_s32g_flush
-- 
2.17.1

